lib/activerecord/delay_touching.rb

```diff
@@ -76,8 +76,13 @@ def self.apply
 
     # Touch the specified records--non-empty set of instances of the same class.
     def self.touch_records(attr, klass, records)
-      attributes = records.first.send(:timestamp_attributes_for_update_in_model)
+      # Although we're now setting the default timestamp column upstream, we'll still want to grab the default attributes here.
+      # Doing so allows us to batch updates to non-standard columns along with the defaults in one query.
+      # Note: timestamp_attributes_for_create_in_model gets frozen before returning in ActiveRecord version 6+
+      frozen_attributes = records.first.send(:timestamp_attributes_for_update_in_model)
+      attributes = frozen_attributes.dup
       attributes << attr if attr
+      attributes.to_set
```

Comment:
`to_set` doesn't mutate, so this is a no-op
```
development main · > attributes = [1, 2, 3]
=> [1, 2, 3]
development main · > attributes.to_set
=> #<Set: {1, 2, 3}>
development main · > attributes
=> [1, 2, 3]
```

You can either store this, or you can use `attributes.uniq!`. One difference between keeping this a set or an array is that with a set, you lose ordering. Probably fine here, just FYI.

Reply:
Would also be curious if this is something we could have tested for.

Reply:
Good call here! Initially I had added this in the first place to ensure we didn't process the same record more than once. This is un-necessary in the first place due to the persistence check further down in this function anyways, so should be fine to remove as it has no effect anyways